src/DataProtection/DataProtection/src/Internal/ContainerUtils.cs

```diff
@@ -77,14 +77,21 @@ internal static bool IsDirectoryMounted(DirectoryInfo directory, IEnumerable<str
             return false;
         }
 
-        private static bool IsProcessRunningInDocker()
+        private static bool IsProcessRunningInContainer()
         {
+            // Official .NET Core images (Windows and Linux) set this. So trust it if it's there.
+            if(string.Equals(Environment.GetEnvironmentVariable("DOTNET_RUNNING_IN_CONTAINERS"), "true", StringComparison.OrdinalIgnoreCase))
```

Comment:
There's a typo here, instead of `DOTNET_RUNNING_IN_CONTAINERS` it should be `DOTNET_RUNNING_IN_CONTAINER`.

See, e.g. https://github.com/dotnet/dotnet-docker/blob/master/2.1/runtime-deps/stretch-slim/amd64/Dockerfile#L21 for where it is set, or e.g. https://www.hanselman.com/blog/DetectingThatANETCoreAppIsRunningInADockerContainerAndSkippableFactsInXUnit.aspx for another source.

Reply:
Looks like the environment variable isn't consistent... See https://github.com/dotnet/dotnet-docker/blob/4fd3b74fc5b7444f39f3fdcffa45eef467bccb24/2.1/aspnet/nanoserver-1803/amd64/Dockerfile

Reply:
Perhaps we should just check both then :)

Reply:
Never mind, I see that the plural is deprecated. Will update!